src/App.jsx

```diff
@@ -14,45 +17,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_TYPE_LENG = 'leng';
+const SORT_TYPE_ALPH = 'alph';
+const REVERSE_BASE_STATE = false;
+
+const updateGoods = (goods, { sortType, isReverse }) => {
+  const udatedGoods = [...goods];
+
+  if (sortType) {
+    udatedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SORT_TYPE_LENG:
+          return good1.length - good2.length;
+        case SORT_TYPE_ALPH:
+          return good1.localeCompare(good2);
+        default:
+          return 0;
```

Comment:
This looks not optimal

You do have beautiful code, but that solution still requires running the sort even when it is not required

I would suggest changing the code n a way you don't run the sort, unless you need it